### ISSUE TITLE
Adapt "public link" permissions to oC server 10.0.4 UI

### ISSUE THREAD

**Issue by michaelstingl** [owncloud/android#2066]
UI for "public link" permissions has been improved with oC server 10.0.4:
https://github.com/owncloud/core/pull/29376

![bildschirmfoto vom 2017-10-27 16-30-06](https://user-images.githubusercontent.com/12717530/32109172-3460e192-bb34-11e7-9f91-6f02710c95dc.png)

@davigonz What is the effort to bring this to the Android client?

**Comment by davigonz** [comment]
@michaelstingl This is for public link, not for private links, isn't it? 

I really like the new distribution of sharing permissions in core, it is much clearer for the user.

> What is the effort to bring this to the Android client? 

Well, in terms of UI, we just have to replace the two switch buttons (Allow editing & Show file listing) with a new dropdown menu, or something like that, which includes the new three options. I will create a mockup.

In terms of logic, it shouldn't take too much because we just have to move the same functionality from two buttons to a list with three options. 

So, to sum up, I would say that is easy to implement, one or two days at most.

**Comment by theScrabi** [comment]
This is how it will look like
![screenshot_1521037265](https://user-images.githubusercontent.com/1891273/37408233-ef01c11e-279b-11e8-927a-12b2e4ae9213.png)

**Comment by jesmrec** [comment]
The links to files has also incuded the "Read Only" option:

![screen shot 2018-05-03 at 14 54 18](https://user-images.githubusercontent.com/14894746/39577565-fbc901b0-4ee1-11e8-9c4a-bd0c87a72da7.png)

will we include in the scope of this issue, or create a separate one? @theScrabi @davigonz

**Comment by theScrabi** [comment]
Is the readonly a radio button here or a check box?

**Comment by theScrabi** [comment]
I think this should be in scope of this. However I think this option is not working on server side because it looks like a radio button and it behaves like one, tho it should be a checkbox. This is why clicking on it has no action.

![gurken](https://user-images.githubusercontent.com/1891273/39624866-2d5cd4c2-4f9b-11e8-88c2-277c69f78497.gif)

However maybe it's not intended to make a single file share writable.

**Comment by jesmrec** [comment]
let's check with core to take a decision.

**Comment by pmaier1** [comment]
> However I think this option is not working on server side because it looks like a radio button and it behaves like one, tho it should be a checkbox

Just to clarify: It works as expected and should not be a checkbox as it's the only available permission currently for public links on single files. We will add read/write permission to the server at some point but not yet scheduled. Clients should adapt to that when it's available.

**Comment by michaelstingl** [comment]
okay, thanks @pmaier1 . Then let's go for the next 2.8.0 release…
